### PR TITLE
Add test for disconnecting ems on targeted refresh

### PR DESCRIPTION
**Pull request by borod108**
Test that when a template does not exist on the provider a targeted refresh
will disconnect it from the ems.

This depends on: https://github.com/ManageIQ/manageiq/pull/16718

### CODE REVIEW

**Comment by borod108** [comment]
@masayag, @pkliczewski  can you please review?

File: spec/models/manageiq/providers/redhat/infra_manager/refresh/refresher_target_template_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    end
+
+    it 'removes connection of ems if template is deleted on provider' do
+      EmsRefresh.refresh(@ems)
+      @ems.reload
+      template = VmOrTemplate.where(:name => "tfail11").first
+      # At this point the template was deleted from the provider
+      expect(template.ems_id).to eq(@ems.id)
+      EmsRefresh.refresh(template)
+      expect(template.reload.ems_id).to be_nil
```
Comment (Ladas) [inline spec/models/manageiq/providers/redhat/infra_manager/refresh/refresher_target_template_spec.rb]: looks like this doesn't pass

Comment (borod108) [reply spec/models/manageiq/providers/redhat/infra_manager/refresh/refresher_target_template_spec.rb]: it depends on https://github.com/ManageIQ/manageiq/pull/16718
locally for me it passes when above mentioned PR is applied.

Comment (pkliczewski) [reply spec/models/manageiq/providers/redhat/infra_manager/refresh/refresher_target_template_spec.rb]: so let's have the dept merged first and let's rerun the tests

**Comment by agrare** [comment]
Dependent PR merged, kicking tests

**Comment by borod108** [comment]
Depends on: https://github.com/ManageIQ/manageiq/pull/16732

**Comment by borod108** [comment]
[[Target Refresh] Removed Template is not updated on CFME UI](https://trello.com/c/m3x9p6lv/289-target-refresh-removed-template-is-not-updated-on-cfme-ui)

**Comment by borod108** [comment]
[Bug 1511445 - Template added to CFME even though sealing fails in RHV](https://trello.com/c/SkwV2TwX/267-bug-1511445-template-added-to-cfme-even-though-sealing-fails-in-rhv)

**Comment by borod108** [comment]
@miq-bot add_label gaprendashvili/yes

**Comment by borod108** [comment]
@miq-bot add_lable gaprindashvili/yes

**Comment by pkliczewski** [comment]
@miq-bot add_label gaprindashvili/yes

File: app/models/manageiq/providers/redhat/infra_manager/refresh/refresher.rb [hunk 2]
```diff
@@ -34,6 +34,8 @@ def collect_inventory_for_targets(ems, targets)
           end
         end
 
+        empty_hash_when_target_not_found!(target, data)
```
Comment (pkliczewski) [inline app/models/manageiq/providers/redhat/infra_manager/refresh/refresher.rb]: How that would work when a `target` is ems?

Comment (masayag) [reply app/models/manageiq/providers/redhat/infra_manager/refresh/refresher.rb]: `corresponding_hashes_empty?` should take care of skipping the ems, but that can be more explicit

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 560859225ab2d2561f6b0e50291cccf24d3bd813
Author: Moti Asayag <masayag@redhat.com>
Date:   Mon Jan 15 14:11:06 2018 +0200

    Merge pull request #184 from borod108/bugs/dissconnect_targeted_template
    
    Add test for disconnecting ems on targeted refresh
    (cherry picked from commit 455b3f22ec8c608acd6b8965ae74f01254159781)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1532648
```

**Pull request merged**